test/utils.ts

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+export const mockStore = {
+  getState: () => ({
+    opensearch: {
+      errorMessage: '',
+    },
+    workflows: {
+      loading: false,
+      errorMessage: '',
+      workflows: {
+        '12345': {
+          id: '12345',
+          name: 'test_workflow',
+          use_case: 'CUSTOM',
+          description: 'A blank workflow with no preset configurations',
+          version: { template: '1.0.0', compatibility: ['2.17.0', '3.0.0'] },
+          workflows: {},
+          ui_metadata: {
+            type: 'Custom',
+            config: {
+              search: {
+                pipelineName: {
+                  id: 'pipelineName',
+                  type: 'string',
+                  value: 'search_pipeline_248e2f68b43db682',
+                },
+                request: {
+                  id: 'request',
+                  type: 'json',
+                  value:
+                    '{\n  "query": {\n    "match_all": {}\n  },\n  "size": 1000\n}',
+                },
+                index: { name: { id: 'indexName', type: 'string' } },
+                enrichRequest: { processors: [] },
+                enrichResponse: { processors: [] },
+              },
+              ingest: {
+                pipelineName: {
+                  id: 'pipelineName',
+                  type: 'string',
+                  value: 'ingest_pipeline_7b139fd4eccac336',
+                },
+                enrich: { processors: [] },
+                index: {
+                  settings: { id: 'indexSettings', type: 'json' },
+                  mappings: {
+                    id: 'indexMappings',
+                    type: 'json',
+                    value: '{\n  "properties": {}\n}',
+                  },
+                  name: {
+                    id: 'indexName',
+                    type: 'string',
+                    value: 'my-new-index',
+                  },
+                },
+                enabled: { id: 'enabled', type: 'boolean', value: true },
+              },
+            },
+          },
+          resourcesCreated: [],
+        },
+      },
+    },
+  }),
+  dispatch: jest.fn(),
+  subscribe: jest.fn(),
+  replaceReducer: jest.fn(),
+  [Symbol.observable]: jest.fn(),
+};
+
+global.ResizeObserver = class {
+  observe() {}
+  unobserve() {}
+  disconnect() {}
+};
```

Comment:
just to double check, since I'm not super familiar - does the `global.*` create any issues when running code in public/ or server/? Want to make sure this doesn't interfere / override any functionality when the plugin is running outside of the test env.

Reply:
Replacing it with below mock. 
```
const resizeObserverMock = jest.fn().mockImplementation(() => ({
  observe: jest.fn(),
  unobserve: jest.fn(),
  disconnect: jest.fn(),
}));

window.ResizeObserver = resizeObserverMock;

```